impala/hiveserver2.py

```diff
@@ -1112,37 +1143,43 @@ def __init__(self, service, handle, config, hs2_protocol_version,
 
     def close(self):
         req = TCloseSessionReq(sessionHandle=self.handle)
-        self._rpc('CloseSession', req)
+        # CloseSession rpcs don't need retries since we catch all exceptions and close
```

Comment:
The close function doesn't seem to handle exceptions. Are we catching the exceptions and closing the transport in a top level caller somewhere?

Reply:
There is no top level handler so the comment is a lie (copied from shell).
I think CloseSession would be non-idempotent as you cannot close a session twice
https://github.com/apache/impala/blob/4099a606892c377b9e8c9c6df2a45a7d42afcaea/tests/hs2/test_hs2.py#L203
I will fix the comment but I think closing the transport is a non-trivial change.